atom/browser/api/atom_api_session.cc

```diff
@@ -224,29 +224,27 @@ void ResolveOrRejectPromiseInUI(util::Promise promise, int net_error) {
 // Callback of HttpCache::GetBackend.
 void OnGetBackend(disk_cache::Backend** backend_ptr,
                   Session::CacheAction action,
-                  const util::CopyablePromise& promise,
+                  util::Promise promise,
                   int result) {
   if (result != net::OK) {
     std::string err_msg =
         "Failed to retrieve cache backend: " + net::ErrorToString(result);
-    util::Promise::RejectPromise(promise.GetPromise(), std::move(err_msg));
+    util::Promise::RejectPromise(std::move(promise), std::move(err_msg));
   } else if (backend_ptr && *backend_ptr) {
     if (action == Session::CacheAction::CLEAR) {
-      auto success =
-          (*backend_ptr)
-              ->DoomAllEntries(base::BindOnce(&ResolveOrRejectPromiseInUI,
-                                              promise.GetPromise()));
+      auto success = (*backend_ptr)
+                         ->DoomAllEntries(base::BindOnce(
+                             &ResolveOrRejectPromiseInUI, std::move(promise)));
```

Comment:
Wouldn't the storage class `BindState` of `base::Bind` move the parameters into their members even before the callback is called ? If thats the case, trying to access it below will crash. /cc @zcbenz

Reply:
Yeah we can't `std::move(promise)` twice.

According to the source code of `DoomAllEntries`, the return value is always `ERR_IO_PENDING`, so we can just ignore the return value. The return value is only kept to make old code compile.
https://cs.chromium.org/chromium/src/net/disk_cache/blockfile/backend_impl.cc?sq=package:chromium&g=0&l=1284